DEVDOCS-4847: [update] Building Storefront Channels, update protected UI settings (#1937)
bigcommerce/dev-docs: docs/api-docs/channels/channels-storefront-tutorial.mdx

## Creating a channel

> After receiving the final response of the [single-click app OAuth flow](/api-docs/apps/guide/auth#receiving-the-post-response) `auth` callback, create a channel on the merchant's store. This allows merchants to assign product listings to the storefront and configure storefront specific settings. To create a storefront channel, set `type` to `storefront` in a request to the [Create a channel](/docs/rest-management/channels#create-a-channel) endpoint.

Reviewer: ```suggestion
After receiving the final response of the [single-click app OAuth flow](/api-docs/apps/guide/auth#receiving-the-post-response) `auth` callback, create a channel on the merchant's store. This channel allows merchants to assign product listings to the storefront and configure storefront-specific settings. To create a storefront channel, set `type` to `storefront` in a request to the [Create a channel](/docs/rest-management/channels#create-a-channel) endpoint.
```

> After receiving the final response of the [single-click app OAuth flow](/api-docs/apps/guide/auth#receiving-the-post-response) `auth` callback, create a channel on the merchant's store. This allows merchants to assign product listings to the storefront and configure storefront specific settings. To create a storefront channel, set `type` to `storefront` in a request to the [Create a channel](/docs/rest-management/channels#create-a-channel) endpoint.

Author: ```suggestion
After receiving the final response of the [single-click app OAuth flow](/api-docs/apps/guide/auth#receiving-the-post-response) `auth` callback, create a channel on the merchant's store. Creating a dedicated channel for a storefront allows merchants to assign the storefront selected product listings and configure storefront-specific settings. To create a storefront channel, set `type` to `storefront` in a request to the [Create a channel](/docs/rest-management/channels#create-a-channel) endpoint.
```

```

<Callout type="info">

>   These UI sections were previously managed using `config_meta`. As of March 29, 2022, changes to UI sections that use `config_meta` are dual-written to the Channel Menus object when `config_meta.app.sections` exists in the payload.

Author: ```suggestion
  Previously, you could manage these settings using the `config_meta` property on the Channels object. As of March 29, 2022, changes to UI sections for channels that already have a non-null `config_meta.app.sections` property are dual-written to the Channel Menus object.
```

>   These UI sections were previously managed using `config_meta`. As of March 29, 2022, changes to UI sections that use `config_meta` are dual-written to the Channel Menus object when `config_meta.app.sections` exists in the payload.

Author: will remove the passive voice.

</Callout>

## Protected UI sections

BigCommerce allows the developer to toggle the location and scope of the following BigCommerce-managed settings pages between the storewide menu and channel-specific menus:

| Menu item / settings page | API Argument | Description |
|:------|:------|:------------|
| Storefront Settings | `storefront_settings` | Renders channel-specific storefront settings. |
| Carousel | `carousel` | Renders channel-specific carousel settings. |
| Currencies | `currencies` | Renders channel-specific currency settings. |
| Domains | `domains` | Renders channel-specific domain settings. |
| Notifications | `notifications` | Renders channel-specific notification settings. |
| Social | `social` | Renders channel-specific social media settings. |

> The term _protected section_ refers to the settings in that section being protected from the settings selected at the storewide level. Include these protected sections in requests to the [Create channel menus](/docs/rest-management/channels/channel-menus#create-channel-menus) endpoint to display BigCommerce-provided, channel-specific settings pages and corresponding menu items. Enabled protected sections precede custom sections in a channel's menu.

Reviewer: ```suggestion
The term _protected section_ refers to secure areas that are protected from storewide-level selections. Include these protected sections in requests to the [Create channel menus](/docs/rest-management/channels/channel-menus#create-channel-menus) endpoint to display BigCommerce-provided, channel-specific settings pages and corresponding menu items. Enabled protected sections precede custom sections in a channel's menu.
```

Author: this is a bit word salady, isn't it. thanks, will revise.

> The term _protected section_ refers to the settings in that section being protected from the settings selected at the storewide level. Include these protected sections in requests to the [Create channel menus](/docs/rest-management/channels/channel-menus#create-channel-menus) endpoint to display BigCommerce-provided, channel-specific settings pages and corresponding menu items. Enabled protected sections precede custom sections in a channel's menu.

Author: ```suggestion
The term _protected section_ refers to the settings on the corresponding menu pages. Settings in protected sections override the default settings configured at the storewide level. Include these protected sections in requests to the [Create channel menus](/docs/rest-management/channels/channel-menus#create-channel-menus) endpoint to display BigCommerce-provided, channel-specific settings pages and corresponding menu items. Enabled protected sections precede custom sections in a channel's menu.
```

}
```

![Protected Sections](https://storage.googleapis.com/bigcommerce-production-dev-center/images/channels/channels-sf-protected-custom-settings.png "Protected Sections")

## Storefront settings

Include the `storefront_settings` [protected section](#protected-ui-sections) in a Channel Menu's `bigcommerce_protected_app_sections` array to render the BigCommerce-provided **Storefront Settings** menu item in the channel's navigation and corresponding settings page, and protect channel storefront settings from those set for the store in general.

![Channel Storefront Settings](https://storage.googleapis.com/bigcommerce-production-dev-center/images/channels/channels-sf-storefront-settings.png "Channel Storefront Settings")

> Read channel specific storefront settings using the [Settings API](/docs/rest-management/settings). For example, you can see a storefront's category settings using the [Get storefront category settings](/docs/rest-management/settings/storefront-category#get-storefront-category-settings) endpoint.

Reviewer: ```suggestion
Read channel-specific storefront settings using the [Settings API](/docs/rest-management/settings). For example, you can see a storefront's category settings using the [Get storefront category settings](/docs/rest-management/settings/storefront-category#get-storefront-category-settings) endpoint.
```
